tests/kernel/ee/interrupt/dmac_order.cpp

```diff
@@ -0,0 +1,169 @@
+#include <common-ee.h>
+#include <kernel.h>
+#include <ee_regs.h>
+#include <string.h>
+
+#define MAX_DMAC_RESULTS 9
+#define MAX_DMAC_HANDLERS 10
+
+static u32 g_dmacSema = 0;
+static u32 g_dmacHandlers[MAX_DMAC_HANDLERS];
+static u32 g_dmacHandlerCount = 0;
+static u32 g_orderPos = 0;
+static u32 g_order[MAX_DMAC_RESULTS];
+
+static const u32 __attribute__((aligned(16))) g_dummyTransferData[4] = {0, 0, 0, 0};
+
+typedef s32 (*DmacHandlerFunction)(s32);
+
+void appendResult(u32 result) {
+	if(g_orderPos != MAX_DMAC_RESULTS) {
+		iSignalSema(g_dmacSema);
+		g_order[g_orderPos] = result;
+		g_orderPos++;
+	}
+}
+
+template <s32 number>
+s32 continueDmacHandler(s32 channel) {
+	appendResult(number);
+	return 0;
+}
+
+template <s32 number>
+s32 cancelDmacHandler(s32 channel) {
+	appendResult(number);
+	return -1;
```

Comment:
Well, I feel like this is an ordering related question - since this is testing ordering/cancelation already, what values actually cause this cancelation?

I could definitely see some game returning an error code from another function here, one that is maybe 0x8xxxxxxx, and the handler always canceling for negative values... but an emulator specifically checking -1, which is likely to be the most common value by far.

-[Unknown]